packages/web/src/components/ui/SearchResultsSidebar/SearchResultsSidebar.tsx

```diff
@@ -18,117 +18,142 @@ import { Button } from '../index';
+
+  const handleFilterReset = useCallback(() => onFiltersChanged(null), [onFiltersChanged]);
+
+  // TODO: Refactor this through restructuring underlying components properly
+  const {
+    handleTraitsFilterChange,
+    handleTraitsFilterReset,
+    handleKeywordsFilterChange,
+    handleKeywordsFilterReset,
+    handleAuthorsFilterChange,
+    handleAuthorsFilterReset,
+  } = useMemo(
+    () => ({
+      handleTraitsFilterChange: (newTraits: string[]) =>
+        onFiltersChanged({
+          ...selectedFilters,
+          traits: newTraits as PackageTrait[],
+        }),
+      handleTraitsFilterReset: () =>
+        onFiltersChanged({
+          ...selectedFilters,
+          traits: [],
+        }),
+      handleKeywordsFilterChange: (newKeywords: string[]) =>
+        onFiltersChanged({
+          ...selectedFilters,
+          keywords: newKeywords,
+        }),
+      handleKeywordsFilterReset: () =>
+        onFiltersChanged({
+          ...selectedFilters,
+          keywords: [],
+        }),
+      handleAuthorsFilterChange: (newAuthors: string[]) =>
+        onFiltersChanged({
+          ...selectedFilters,
+          authors: newAuthors,
+        }),
+      handleAuthorsFilterReset: () =>
+        onFiltersChanged({
+          ...selectedFilters,
+          authors: [],
+        }),
+    }),
+    [onFiltersChanged, selectedFilters]
+  );
 
   return (
     <>
-      <Modal show={modalKeywordsOpen} setShow={setModalKeywordsOpen}>
+      <Modal isOpen={activeModal === 'keywords'} onClose={closeModalHandler}>
         <SidebarCategoryWithSearch
           categoryName='Keywords'
-          keywordsList={keyWords}
-          selectedKeywords={selectedKeywords}
-          selectHandler={handleKeywordsChange}
-          returnButton={() => setModalKeywordsOpen(false)}
-          resetGroup={() => setSelectedKeywords([])}
+          keywordsList={availableFilters.keywords}
+          selectedKeywords={selectedFilters.keywords}
+          selectHandler={handleKeywordsFilterChange}
+          returnButton={closeModalHandler}
+          resetGroup={handleKeywordsFilterReset}
           searchOpen
         />
       </Modal>
 
-      <Modal show={modalProblemsOpen} setShow={setModalProblemsOpen}>
+      <Modal isOpen={activeModal === 'traits'} onClose={closeModalHandler}>
         <div className={modalStyles.modalContentWrapper}>
           <SidebarCategory
             categoryName='Problems'
-            selectedKeywords={selectedProblems}
-            returnButton={() => setModalProblemsOpen(false)}
-            resetGroup={() => setSelectedProblems([])}
+            selectedKeywords={selectedFilters.traits}
+            returnButton={closeModalHandler}
+            resetGroup={handleTraitsFilterReset}
           >
             <ProblemsList
-              keywordsList={problems}
-              selectedKeywords={selectedProblems}
-              selectHandler={handleProblemsChange}
+              keywordsList={availableFilters.traits}
```

Comment:
Looks like renaming the 'problems' entity to 'traits' was unnecessary. However, my point is - we shouldn't use different terminology for the same entity, so I probably could agree with renaming to 'traits' _everywhere_, but now it's not the case.

Reply:
I see where you're coming from, but distinct names in business logic domain and UI presentation layer aren't a problem per se and encountered from time to time. In this very case I'd still resort to current naming having imminent additional features (like `duplicate`, etc.) in mind.

Reply:
Different naming increases cognitive load and amounts of chaos in the codebase. It's not about logic domain or ui, it's about how you or someone else reads the code and understands its meaning and how different parts are connected to each other. I can approve this if you insist, though it is to be refactored in nearest future.

Reply:
The reasoning you gave was exactly the case why i chose to rename it the current way. "Package trait" or "Package feature" when talking about a "vulnerable" / "outdated" / "duplicate" package makes much more sense to me rather than a "Package problem". @spalt08 Your call.